tools/wpt/browser.py

```diff
@@ -1104,6 +1104,38 @@ def version(self, binary=None, webdriver_binary=None):
         return None
 
 
+class WebKitGTKMiniBrowser(WebKit):
+
+    def find_binary(self, venv_path=None, channel=None):
+        libexecpaths = ["/usr/libexec/webkit2gtk-4.0",
+                        "/usr/lib/x86_64-linux-gnu/webkit2gtk-4.0"]
+        # If GCC is available use it to detect this machine triplet,
+        # and try to look into its multi-arch libexecdir.
+        gcc = find_executable("gcc")
+        if gcc:
+            try:
+                triplet = call(gcc, "-dumpmachine").strip()
+            except subprocess.CalledProcessError:
+                triplet = None
+            if triplet:
+                libexecpaths.insert(0, "/usr/lib/%s/webkit2gtk-4.0" % triplet)
+        return find_executable("MiniBrowser", os.pathsep.join(libexecpaths))
+
+    def find_webdriver(self, channel=None):
+        return find_executable("WebKitWebDriver")
+
+    def version(self, binary=None, webdriver_binary=None):
+        if binary is None:
+            return None
+        try: # WebKitGTK MiniBrowser before 2.26.0 doesn't support --version
+            output = call(binary, "--version").strip()
+        except subprocess.CalledProcessError:
+            output = None
+        if output:
+            return output.split()[1]
```

Comment:
What is the first component? Can you comment on the output format and add a test similar to here:?
https://github.com/web-platform-tests/wpt/blob/master/tools/wpt/browser.py#L983
https://github.com/web-platform-tests/wpt/blob/master/tools/wpt/tests/test_browser.py#L12

Reply:
The output looks like this:
```
WebKitGTK 2.26.1
```
I will add a test.

Reply:
Matching a regexp like in `r"Included with Safari (.*)"` would give some assurance that if the output format changes this stops matching instead of returning some other part of the output that isn't a version numbers.

Reply:
Fixed in https://github.com/web-platform-tests/wpt/pull/19227/commits/462c2dfbf7534d944af4fda1081018f8f6b21f0c.